src/pydap/model.py

```diff
@@ -568,6 +571,20 @@ def __getitem__(self, key):
                 var.data = self[var.name].data[slice_]
             return out
 
+    def __len__(self):
+        """Return the first children length."""
+        return len(self.array)
+
+    @property
+    def dtype(self):
+        """Return the first children dtype."""
+        return self.array.dtype
+
+    @property
+    def shape(self):
+        """Return the first children shape."""
+        return self.array.shape
```

Comment:
It's not strictly necessary, but I usually make my duck array types implement `__array__`, `ndim` and `size` as well:
https://github.com/pydata/xarray/blob/371d034372bc7522098a142a0debf93916c49102/xarray/core/utils.py#L406

(Implement these on `BaseType`, too, if you decide to go this route)